proto/contlcycle/contlcycle.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+option go_package = "github.com/DataDog/agent-payload/v5/contlcycle";
+
+package datadog.contlcycle;
+
+// EventsPayload represents the main container lifecycle event payload
+message EventsPayload {
+    enum ObjectKind {
+        Cont = 0;
+        Pod = 1;
+    }
+    string version = 1; // version represents the EventsPayload message version
+    ObjectKind objectKind = 2; // objectKind represents the object kind (container, pod)
+    string host = 3; // host contains the hostname
+    repeated Event events = 4; // events contains all the event details
+}
+
+// Event contains the details about a container lifecycle event
+message Event {
+    enum EventType {
+        Delete = 0;
+    }
+    EventType eventType = 1; // eventType represents the event's type (deletion)
+    string objectID = 2; // objectID contains the container/pod ID
+    string source = 3; // source source contains the container/pod source (docker, kubelet..)
+    oneof optionalExitCode {
+        int32 exitCode = 4;  // exitCode contains the exit code of the container
+    }
+    oneof optionalExitTimestamp {
+        int64 exitTimestamp = 5; // exitTimestamp contains the exit timestamp of the container
+    }
```

Comment:
I'm wondering if we should create 2 Sub-Event messages to clearly identify which field will be linked to container, and the other to pod

```suggestion
    oneof typedEvent {
        ContainerEvent container = 2
        PodEvent pod = 3
    }
}

message ContainerEvent {
    string containerID = 2; // containerID contains the container/pod ID
    string source = 3; // source source contains the container/pod source (docker, kubelet..)
    oneof optionalExitCode {
        int32 exitCode = 4;  // exitCode contains the exit code of the container
    }
    oneof optionalExitTimestamp {
        int64 exitTimestamp = 5; // exitTimestamp contains the exit timestamp of the container
    }
    

message PodEvent {
    string podUID = 1; // containerID contains the container/pod ID
    
    oneof optionalTerminatedStatus {
        string terminatedStatus = 2;  // terminatedStatus can be succeed, failed for Job's pod.
    }
    oneof optionalTerminatedStatusReason {
        string terminatedReason = 3; // terminatedReason contains the failure reason or the terminated reason like "evicted"
    }
}
```

Reply:
Yes we can do this, I'm just hesitant about the pods `terminatedStatus` and `terminatedReason`. Currently workloadmeta doesn't provide this info, and potential consumers don't need it - I'd prefer adding these to the payload only when a consumer need them. I'd keep only `podUID` in `PodEvent` for now.

Reply:
yes for `terminatedStatus`  and `terminatedReason` it can be added later.